src/js/Expression.js

```diff
@@ -68,13 +70,61 @@ export class Expression {
             if (Expression.#isDigit(this.#input[i])) {
                 let number = this.#parseNumberAt(i)
                 i += Expression.#lengthOf(number) - 1
+                this.#validateTokenOrder(number)
                 yield number
-            } else if (this.#input[i] === " ") {
-                continue
-            } else {
+            } else if (isOperator(this.#input[i]) || isBracket(this.#input[i])) {
+                this.#validateTokenOrder(this.#input[i])
                 yield this.#input[i]
+            } else {
+                throw new InvalidInput(this.#input)
```

Comment:
Did we decide to change something about the `InvalidInput` error in `Operators.apply`? I don't remember exactly.